### PR TITLE
Added functionality to let us join an already existing cluster.

### PR DESCRIPTION
**Pull request by oliverisaac**
**What this PR does / why we need it**: If you are setting up a multi-cluster cockroach deployment this PR allows you to specify an already existing cluster to join. This also lets you specify a locality for your instances so that cockroachdB handles the multi-cluster configuration correctly.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: stable/cockroachdb/templates/cockroachdb-statefulset.yaml [hunk 1]
```diff
@@ -288,7 +290,7 @@ spec:
...
+          - "exec /cockroach/cockroach start --logtostderr {{ if .Values.Secure.Enabled }}--certs-dir /cockroach/cockroach-certs{{ else }}--insecure{{ end }} --advertise-host $(hostname).${STATEFULSET_FQDN} --http-host 0.0.0.0 --http-port {{ .Values.InternalHttpPort }} --port {{ .Values.InternalGrpcPort }} --cache {{ .Values.CacheSize }} --max-sql-memory {{ .Values.MaxSQLMemory }} {{ if .Values.Locality }}--locality={{.Values.Locality }}{{ end }} --join {{ if .Values.JoinExisting }}{{ join "," .Values.JoinExisting }}{{ else }}${STATEFULSET_NAME}-0.${STATEFULSET_FQDN}:{{ .Values.InternalGrpcPort }},${STATEFULSET_FQDN}-1.${STATEFULSET_FQDN}:{{ .Values.InternalGrpcPort }},${STATEFULSET_FQDN}-2.${STATEFULSET_FQDN}:{{ .Values.InternalGrpcPort }}{{ end }}"
```
Comment (a-robinson) [inline stable/cockroachdb/templates/cockroachdb-statefulset.yaml]: Your join list looks wrong. It shouldn't be `${STATEFULSET_FQDN}-1.${STATEFULSET_FQDN}` and `${STATEFULSET_FQDN}-2.${STATEFULSET_FQDN}`, it should be `${STATEFULSET_NAME}-1.${STATEFULSET_FQDN}` and `${STATEFULSET_NAME}-2.${STATEFULSET_FQDN}`

File: stable/cockroachdb/values.yaml [hunk 2]
```diff
@@ -62,3 +62,9 @@ Secure:
     # The name of the service account to use.
     # If not set and create is true, a name is generated.
     Name:
+
+# If you are deploying a second cockroach instance that should join a first, use the below list to join to the existing instance.
+# Each item in the array should be a FQDN (and port of needed) resolvable by the new pods
```
Comment (a-robinson) [inline stable/cockroachdb/values.yaml]: s/of/if/

**Review by a-robinson — COMMENTED** [review]
Nice addition!

File: stable/cockroachdb/templates/cockroachdb-statefulset.yaml (hunk 1 above)
Comment (oliverisaac) [reply stable/cockroachdb/templates/cockroachdb-statefulset.yaml]: Oh man, I got lucky that joining -0 was working so it didn't have the opportunity to totally blow up. Good catch!

**Comment by oliverisaac** [comment]
Fixed up those issues, should be good to go now!

**Pull request merged**